src/python/grapl-common/grapl_common/envelope.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import uuid
+from uuid import uuid4
+
+from graplinc.grapl.api.services.v1beta1.types_pb2 import Envelope as _Envelope
+from graplinc.grapl.api.services.v1beta1.types_pb2 import Meta, ProtoUuidV4
+
+
+def proto_uuid_to_pyuuid(proto_uuid: ProtoUuidV4) -> uuid.UUID:
```

Comment:
These methods feel like they want to be `Uuid::from(uuid: ProtoUuidV4) -> Uuid` and `Uuid::into() -> uuid.UUID`, which would be nice.

Reply:
Yes but you can't do that in Python.

Reply:
Sorry, to be clear I'm talking about writing python types like this:

```python
class Uuid:

    def __init__(self, uuid: uuid.UUID) -> Uuid:
        self._uuid = uuid

    @staticmethod
    def from(uuid: ProtoUuidV4) -> Uuid:
        ...

    def into() -> uuid.UUID:
        ...

    def into_proto() -> ProtoUuidV4:
        ...
```
and
```python
class Metadata:

    def __init__(self, tenant_id: Uuid, trace_id: Uuid):
        self.tenant_id = tenant_id
        self.trace_id = trace_id

    @staticmethod
    def from(metadata: Meta) -> Metadata:
        ...

   def into_proto() -> Meta:
        ...
```

Reply:
Oh for sure. Yeah, it just gets really awkward really fast. You either have to upfront convert everything, and then serialize back, or you need to have conversion code everywhere.